desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py

```diff
@@ -266,12 +266,20 @@ def explain(self, notebook, snippet):
     engine = self._get_engine()
     connection = engine.connect()
     statement = snippet['statement']
+    result = engine.execute('EXPLAIN '+ statement)
+    result = {'result': dict(row) for row in result}
+    result = result['result']
 
     explanation = ''
+    for key in result:
+      if result[key] == None:
+        explanation += key + ': ' + 'None' + '\n'
+      else:
+        explanation += key + ': ' + str(result[key]) + '\n'
```

Comment:
This could be a one liner, we only care about joining the result rows into one big string here, similarly to https://github.com/cloudera/hue/blob/master/apps/beeswax/src/beeswax/server/hive_server2_lib.py#L1397

Reply:
type of Result is 'sqlalchemy.engine.result.ResultProxy' and this is 'Tuple'. So following this https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/connectors/rdbms.py#L183. And some key's also None so I have to separate them also.